react-app/ccf/src/App.tsx

```diff
@@ -1,10 +1,71 @@
-import React from 'react';
-import logo from './logo.svg';
+import  { useState, ChangeEvent, useEffect } from 'react';
 import './App.css';
+import { handleUpload, listAndDownloadPDFs } from './functions/storage';
 
 function App() {
+  const [file, setFile] = useState<File | null>(null);
+  const [uploadStatus, setUploadStatus] = useState<string>('');
+  const [downloadStatus, setDownloadStatus] = useState<Array<{ name: string, url: string }>>([]);
+
+  useEffect(() => {
+    const fetchPDFsOnLoad = async () => {
+      try {
+        const files = await listAndDownloadPDFs();
+        setDownloadStatus(files);
+      } catch (error) {
+        console.error('Error listing files:', error);
+      }
+    };
+    fetchPDFsOnLoad();
+  }, []);
+
+
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files[0]) {
+      setFile(e.target.files[0]);
+    }
+  };
+
+  const onUpload = async () => {
+    if (!file) {
+      setUploadStatus('Please select a file first!');
+      return;
+    }
+
+    try {
+      setUploadStatus('Uploading...');
+      const result = await handleUpload(file);
+      setUploadStatus(result);
+      setFile(null);
+      try {
+        const files = await listAndDownloadPDFs();
+        setDownloadStatus(files);  
+      } catch (error) {
+        console.error('Error listing files:', error);
+      }
+
+    } catch (error) {
+      console.error('Upload failed', error);
+      setUploadStatus('Upload failed. Please try again.');
+    }
+  };
+
   return (
     <div className="App">
+      <p>Cloud Storage Demo (Ryan and Tara):</p>
```

Comment:
Love the demo for testing, but we should probably remove it before merging this branch to main.